domains/pallets/domain-registry/src/lib.rs

```diff
@@ -958,6 +975,13 @@ impl<T: Config> Pallet<T> {
         DomainCreators::<T>::insert(domain_id, who, deposit);
         NextDomainId::<T>::put(domain_id + 1);
 
+        let current_block_number = frame_system::Pallet::<T>::block_number();
+        CreatedAt::<T>::insert(domain_id, current_block_number);
+        pallet_receipts::Pallet::<T>::initialize_head_receipt_number(
+            domain_id,
+            current_block_number,
+        );
```

Comment:
This may also imply that the node of the newly added core domain needs to process the primary block starting from `current_block_number + 1` or `ReceiptError::MissingParent` may return.

Reply:
Nope, this implies that the receipts will be submitted from `current_block_number + 1` (`created_at + 1`), which is expected and correct because a domain block/receipt is a thing only after this domain has been created, it won't run into `ReceiptError::MissingParent`.

Currently, the newly created domains have to also process the prior blocks at the moment, which is a kind of no-op for the new domain. In the future, I think it's nice to support the new core domain to start processing the primary block after the domain creation, but that requires more work and might have some open questions.

Reply:
> Currently, the newly created domains have to also process the prior blocks at the moment, which is a kind of no-op for the new domain.

Do you mean processing the prior primary block and starting from the genesis block?

Reply:
I meant currently the new domains have to run the embedded primary node from the genesis and process each primary block even before the domain creation so that the new domains can have a proper initial state, in specific, the new domain has the same block number as the primary chain and the genesis domain even this domain created later is not existent until block 100 (assuming it's created at block 100).

Reply:
Okay, that answers my question, thanks!